Widebury

Sai-Aye opened the pull request:
## About The Pull Request

New ship map for crash / zombie crash. Intended for medium crash pop (>=10 and <25).

SDMM Screenshot:
<img width="744" height="680" alt="image" src="https://github.com/user-attachments/assets/ce2b2146-fac7-4497-ae10-094feffde253" />

In-Game Screenshot:
<img width="617" height="565" alt="image" src="https://github.com/user-attachments/assets/e56788ff-7f6e-402f-99bf-604db4d53d5d" />


Notable features include:
- North exit. 
- E/W exits are positioned on the south side of the ship.
- Central cockpit/cic placement
- Motorbike, jerry cans, extra fuel tank in engineering.
- Newt (medbay scug)
- 21 x 23 size

Other Changes:
- Made additions to shuttle.dm and crash.dm to make widebury usable.
- Added a pop requirement in crash.dm to spawn widebury at mid pop.
- Adjusted stationary dock size to fit widebury.
- Fixed a typo in crash.dm 

LZ testing screenshots:
https://docs.google.com/document/d/1EwKMZgSf6NCCt82buF4PNPyiNdS7zfVyQQs2Uu4XkFA/edit?usp=sharing

## Why It's Good For The Game

A new ship map for crash will be a fun addition to a stale gamemode. 
Widebury is more spacious to accommodate mid pop while being smaller than bigbury, and therefore easier to defend/manage.

## Changelog

:cl: Scav
add: Added Widebury - a mid-sized ship map for crash / zombie crash.
/:cl:

Sai-Aye committed:
WIDEbury

Sai-Aye committed:
shuttle stuff

added widebury entries to various shuttle things

fixed a typo in crash.dm

minor widebury map edit

Sai-Aye committed:
added windows

Sai-Aye committed:
cic windows, motorbike

replaced cic hull windows with tough

added motorbike, jerry cans, extra fueltank

Sai-Aye committed:
pop requirement, minor map edits

Sai-Aye committed:
minor wall fixes

Sai-Aye committed:
prep decals

Lumipharon commented:
Have you tested this to make sure it actually fits in all the crash spots? Its so much wider it may have map clipping issues.

The medbay is horrifically cramped and impractical, its a 1 tile maint corridor across the entire area, it will be a nightmare to use.

mudzbe commented:
Can you move the north exit up by one at least, so we have to make less side cades

Sai-Aye commented:
> Isnt the north shutter going to require a lot more barricades and in consequence mats? You either have to build the barricade rows extra wide or you will need to build multiple vertical lines on top of horizontal lines so that zombies cant just break one to get access to the ship. On the standard ship you need to place 27 barricades so that the zombies need to break through 4 until they have access to the ship. The most efficient set up I can see here will need 45 barricades placed to have 4 layers between the outside and inside. You could maybe compensate this by spawning more mats in? Although its still going to be more annoying to set up. 

I added an extra 50 metal and 50 plas. Also moved the north exit up by one tile and added two walls on either side to reduce the number of side cades required.

> I think its looks really cool, though maybe for better balance there should be more of an upside this ship has that you get for trading in the more compact layout of the original ship?

The main upside to this ship are the large external windows and central cic windows that allow for lasgun / bas laser defence setups. I don't know if this fully compensates for the less compact layout, but I'm hoping the extra mats will add to the list of upsides.

Here is a screenshot of the new layout
<img width="742" height="678" alt="image" src="https://github.com/user-attachments/assets/1c0767fc-3024-48c4-a466-a6f985baf1d9" />

Sai-Aye commented:
> Have you tested this to make sure it actually fits in all the crash spots? Its so much wider it may have map clipping issues.

Yes I've used the shuttle manipulator to test all LZs in each crash map. The ship fits into most LZs with only one exit ever being fully blocked off by terrain walls. On LZs that clip into buildings the walls that block an LZ can be C4'd, PC'd, or shot down with MGs. Overall the clipping issues are no worse than what bigbury has to deal with.

> The medbay is horrifically cramped and impractical, its a 1 tile maint corridor across the entire area, it will be a nightmare to use.

The main portion of medbay (surgery, vendors) actually has a 2 tile wide walkable space. The "1 tile maint corridor" is actually just 4 tiles in the north half of medbay. Cryo, chem, and sleeper/autodoc area is more cramped for sure, but I don't think it will be a big issue. Will need to do live testing to know for sure though.

> In general this ship looks very cramped, the key artery of the ship is a 2 wide corridor, with the main part lined with vendors. Its going to be highly congested and a pain to move around IMO, considering the overall size of the ship. This is probably the main reason why just a single large corridor tends to be more practical.

Valid point. To address this I slightly reduced the size of CIC and moved it up by one tile. There is now a 3 wide corridor by prep/vendors and the most cramped areas of the ship are relegated to the 2x5 corridors on either side of CIC. Screenshot below.

<img width="742" height="678" alt="image" src="https://github.com/user-attachments/assets/8dca0047-d47b-4271-bb0b-fe191da0405d" />

Sai-Aye committed:
feedback edits

edited cic size and position
moved north exit up by 1 tile
added 50 extra metal and plasteel

Sai-Aye commented:
> It might already be enough to just switch the white and green areas around so that the 2 tile wide space is in front of the tube and the vendors and surgery are the more cramped section. That would make it about the same as on the Canterbury. 

I tried swapping the sections but it somehow reduced the total free tiles so I scrapped that.

> Or you could just remove the 2 white non-window walls so that both spaces are 2 wide.

This is what I went with in the end. Screenshot below:
<img width="469" height="746" alt="image" src="https://github.com/user-attachments/assets/bca4293c-09ab-4373-957e-9068c244b789" />

> It might also be a good idea to plan for #18497 getting merged and already reserve a spot for the extra vendor so that it doesn't have to be crammed into an awkward spot.

Good point. There is a loadout vendor inside the cockpit that is kinda redundant. I put it there to fill space. It can be replaced with the extra vendor when it gets merged. Screenshot:
<img width="961" height="720" alt="image" src="https://github.com/user-attachments/assets/dbc9637c-5280-4ce7-bba1-d3739095916c" />

Sai-Aye committed:
medbay edit for more free space

spookydonut commented on code/genesis_call.dme:
```diff
@@ -40,3 +40,12 @@
+// BEGIN_PREFERENCES
+// END_PREFERENCES
+// BEGIN_INCLUDE
+// END_INCLUDE
```
what the heck is this

Sai-Aye replied:
Not sure, it kinda auto generated after I made changes to the docking port stuff. I can delete if you want.

Sai-Aye committed:
Delete code/genesis_call.dme

oops

Sai-Aye committed:
Revert "Delete code/genesis_call.dme"

This reverts commit 02fce77ba178f122833ef9aeb29aebe1b94f3b18.

Sai-Aye committed:
deleted stuff

spookydonut reviewed (approved):
shuttle changes appear ok, get whoever else to review balance/mapping

RipGrayson reviewed (changes requested):
I second what Technical-I said. The landmarks were made with the assumption that the Cant was narrow and wouldn't crush disks. Have you verified that the Widebury is safe?

Sai-Aye committed:
more feedback edits

moved around stuff in medbay
rearranged upper prep vendors
added tiles under engines

replaced 1 count phoron stacks with medium stacks in chem
replaced one 25 count plasteel stack with 50 count

removed the loadout vendor in cockpit to make space for req vendor

moved some dropship seats to better positions

mudzbe commented:
You need to add the zcrash vendor and remove the cades now

Sai-Aye committed:
Removed prebuilt cades, added mats

Sai-Aye committed:
Merge remote-tracking branch 'upstream/master' into widebury

Sai-Aye committed:
Added zcrash req vendor

also deleted the second fueltank in engi

Sai-Aye committed:
blocked exits fix (wall clear)

Sai-Aye committed:
removed debug message

Lumipharon commented on code/game/turfs/open.dm:
```diff
@@ -388,3 +388,37 @@
+
+	for(var/obj/O in src)
+		qdel(O)
+	ChangeTurf(floor_path)
```
I already told you we have code for this. Please don't add redundant code and hacky ways of doing things, this is causing issues and failing checks.

Sai-Aye replied:
I asked you on discord whether I should use my own code or replace it with the existing stuff. You did not respond, so I went with my code.

I'll use the existing code now. Give me a couple days to figure it out.

Sai-Aye committed:
Revert "removed debug message"

This reverts commit ad6f8b7e3ae572328e7db9de1f0cfc78fbb77e4e.

Sai-Aye committed:
Revert "blocked exits fix (wall clear)"

This reverts commit 1d0f288dda0b4357a6008b8be5a01b5a189a2914.

larndey commented:
Well, it has been a few months since may eleventh. Shall we keep it?

mudzbe commented:
needs sentry landmarks for https://github.com/tgstation/TerraGov-Marine-Corps/pull/18826

mudzbe commented:
> Well, it has been a few months since may eleventh. Shall we keep it?

> [<img alt="@TiviPlus" width="20" height="20" src="https://avatars.githubusercontent.com/u/57223640?s=40&amp;u=6d8ad4c2dc01ce15ea33c502231d47c12d9d1ef1&amp;v=4">](/TiviPlus) [TiviPlus](/TiviPlus) added the [ Test Merge Candidate ](/tgstation/TerraGov-Marine-Corps/issues?q=state%3Aopen%20label%3A%22Test%20Merge%20Candidate%22) label [Dec 23, 2025](#event-21708055279)

It's been longer, I just realised its been 9 months
